### PR TITLE
feat(hive): Only build and ship Hive metastore

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Needs https://github.com/stackabletech/hive-operator/pull/447

This should also fix the CI

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [x] Changes are OpenShift compatible
- [x] All added packages (via microdnf or otherwise) have a comment on why they are added
- [x] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [x] All packages should have (if available) signatures/hashes verified
- [ ] Does your change affect an SBOM? Make sure to update all SBOMs
- [x] Add an entry to the CHANGELOG.md file
- [x] Integration tests ran successfully
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

File: .github/workflows/dev.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - main
+  workflow_dispatch:
```
Comment (sbernauer) [inline .github/workflows/dev.yaml:7]: All changes in this file are only temporary

**Comment by lfrancke** [comment]
This still includes temporary stuff

**Comment by sbernauer** [comment]
Reverted them, but now we can't test CI any more :yum:

File: conf.py [hunk 2]
```diff
@@ -143,6 +143,10 @@
                 "java-base": "1.8.0",
                 "hadoop": "3.3.4",
                 "jackson_dataformat_xml": "2.12.3",
+                # Normally Hive 3.1.3 ships with "postgresql-9.4.1208.jre7.jar", but as this so old it does only support
```
Comment (Maleware) [inline conf.py:146]:
```suggestion
                # Normally Hive 3.1.3 ships with "postgresql-9.4.1208.jre7.jar", but as this is old enough it does only support
```

File: hive/Dockerfile [hunk 3]
```diff
@@ -122,15 +115,15 @@ RUN microdnf update && \
 USER stackable
 WORKDIR /stackable
 
-COPY --link --from=builder /stackable/apache-hive-${PRODUCT} /stackable/apache-hive-${PRODUCT}
-RUN ln -s /stackable/apache-hive-${PRODUCT}/ /stackable/hive
+COPY --chown=stackable:stackable --from=builder /stackable/apache-hive-metastore-${PRODUCT}-bin /stackable/apache-hive-metastore-${PRODUCT}-bin
```
Comment (Maleware) [inline hive/Dockerfile:118]: Does the build brake with `--link` ? If not I would suggest to keep it as it *should* be more performant

File: hive/Dockerfile [hunk 4]
```diff
@@ -122,15 +115,15 @@ RUN microdnf update && \
...
+RUN ln -s /stackable/apache-hive-metastore-${PRODUCT}-bin/ /stackable/hive-metastore
 
 # It is useful to see which version of Hadoop is used at a glance
 # Therefore the use of the full name here
-COPY --link --from=builder /stackable/hadoop /stackable/hadoop-${HADOOP}
+COPY --chown=stackable:stackable --from=builder /stackable/hadoop /stackable/hadoop-${HADOOP}
```
Comment (Maleware) [inline hive/Dockerfile:123]: Same as above

File: hive/Dockerfile [hunk 5]
```diff
@@ -122,15 +115,15 @@ RUN microdnf update && \
...
 RUN ln -s /stackable/hadoop-${HADOOP}/ /stackable/hadoop
 
-COPY --link --from=builder /stackable/jmx /stackable/jmx
+COPY --chown=stackable:stackable --from=builder /stackable/jmx /stackable/jmx
```
Comment (Maleware) [inline hive/Dockerfile:126]: same as above

File: hive/Dockerfile [hunk 6]
```diff
@@ -139,8 +132,8 @@ COPY hive/licenses /licenses
...
 ENV HADOOP_HOME=/stackable/hadoop
-ENV HIVE_HOME=/stackable/hive
-ENV PATH="${PATH}":/stackable/hadoop/bin:/stackable/hive/bin
+ENV HIVE_HOME=/stackable/hive-metastore
+ENV PATH="${PATH}":/stackable/hadoop/bin:/stackable/hive-metastore/bin
 
-WORKDIR /stackable/hive
-CMD ["./bin/start-metastore", "--config", "conf", "--hive-bin-dir", "bin", "--db-type", "derby"]
+WORKDIR /stackable/hive-metastore
+# Start command is set by oeprator to something like "bin/start-metastore --config /stackable/config --db-type postgres --hive-bin-dir bin"
```
Comment (Maleware) [inline hive/Dockerfile:139]:
```suggestion
# Start command is set by operator to something like "bin/start-metastore --config /stackable/config --db-type postgres --hive-bin-dir bin"
```

File: hive/Dockerfile (hunk 3 above)
Comment (sbernauer) [reply hive/Dockerfile:118]: I can only say the previous version caused failure and the current one seems to be fine.
We might be able to use `COPY --link --chown=stackable:stackable --from=builder /stackable/apache-hive-metastore-${PRODUCT}-bin /stackable/apache-hive-metastore-${PRODUCT}-bin`, but we would need to start the whole CI testing stuff again. Would it be ok to merge this to hopefully get the CI green and than try to improve the performance afterwards if desired?

Comment (Maleware) [reply hive/Dockerfile:118]: yeah! That's allright. I would appreciate a ticket or a comment in the dockerfile to not forget it :)

**Review by Maleware — APPROVED** [review]
Thanks! LGTM

File: hive/Dockerfile (hunk 3 above)
Comment (Maleware) [reply hive/Dockerfile:119]: Thanks :)

**Pull request merged**